File: dashboards/go/handlers/approvals.go
package handlers

import (
	"agenticmail-dashboard/services"
	"agenticmail-dashboard/templates"
	"fmt"
	"net/http"
)

// HandleApprovals renders the approvals page for pending approval requests.
func HandleApprovals(w http.ResponseWriter, r *http.Request) {
	s := services.GetSession(r)

	content := `<h2 class="t">Approvals</h2><p class="desc">Review and manage pending approval requests</p>
<div style="margin-bottom: 20px;">
	<button class="btn btn-p">Pending</button>
	<button class="btn">Approved</button>
	<button class="btn">Rejected</button>
</div>
<div class="card">
	<div class="ct">Pending Approvals</div>
	<div class="empty"><div class="empty-i">✅</div>No pending approvals<br><small>Agent approval requests will appear here</small></div>
</div>
<div class="card">
	<div class="ct">Approval History</div>
	<div class="empty"><div class="empty-i">📋</div>No approval history<br><small>Past approvals and rejections will appear here</small></div>
</div>`

	w.Header().Set("Content-Type", "text/html")
	fmt.Fprint(w, templates.Layout("approvals", s.User, content))
}

File: dashboards/php/pages/approvals.php
<?php
/**
 * Approvals Page — Pending approval requests
 */

layout_start('Approvals', 'approvals');
?>

<h2 class="title">Approvals</h2>
<p class="desc">Review and manage pending approval requests</p>

<div class="tabs">
  <div class="tab active">Pending</div>
  <div class="tab">Approved</div>
  <div class="tab">Rejected</div>
</div>

<div class="card">
  <div class="card-t">Pending Approvals</div>
  <div class="empty">
    <div class="empty-i">✅</div>
    <p>No pending approvals</p>
    <small>Agent approval requests will appear here</small>
  </div>
</div>

<div class="card" style="margin-top: 20px;">
  <div class="card-t">Approval History</div>
  <div class="empty">
    <div class="empty-i">📋</div>
    <p>No approval history</p>
    <small>Past approvals and rejections will appear here</small>
  </div>
</div>

<?php layout_end(); ?>